plottr/plot/mpl.py

```diff
@@ -91,12 +91,15 @@ class PlotType(Enum):
 class ComplexRepresentation(Enum):
     """Options for plotting complex-valued data."""
 
-    #: only real
-    real = auto()
-
+    #: no plot defined
+    empty = auto()
```

Comment:
Rather than having this state that cannot be selected we should programmatically set the button of the default state to checked

Reply:
I removed this and included the default activation.

Reply:
Why is empty back now?

Reply:
Please use the relevant selection between the 3 existing (I would guess real and imag ) for 1d plots and disable the 2 other ones rather than adding empty which still has the problems we discussed last week

Reply:
This empty is because of having the complex options deactivated for non-complex data, which is used in line 959.
To handle having the real/imag as the enabled default, I added this function `def setAllowedComplexTypes(self, *args: ComplexRepresentation)` in line 768. So, in that regard we don't have any problem for both 1D and 2D. We also need this function to handle deactivation complex options for non-complex data.

Reply:
I still do not see why this is needed:

* If there is only real data: select real option and gray out the other ones.
* For 1D plot we show real and imag so select real and imag and gray out the other ones.

Reply:
Point 1: That's fine. I also thought about that and tested. 
Pont 2: I am bit confused why we only want real/imag for 1d plot if the data is complex? We also need to have real, mag/phase for 1D plot, right? Am I missing something?

Reply:
Sorry my bad I assumed that you did what we talked about and disabled complex for 1d plots. If you have actually implemented that we can just use `real` for data that is non complex and remove the empty state